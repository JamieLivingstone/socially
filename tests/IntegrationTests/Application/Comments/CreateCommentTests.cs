using System.Linq;
using System.Threading.Tasks;
using Application.Comments.Commands.CreateComment;
using Application.Common.Exceptions;
using Domain.Entities;
using FluentAssertions;
using IntegrationTests.Application.TestUtils;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace IntegrationTests.Application.Comments
{
  [TestFixture]
  public class CreateCommentTests : TestBase
  {
    [Test]
    public void GivenAnInvalidRequest_ThrowsValidationException()
    {
      var command = new CreateCommentCommand
      {
        Message = "", // Required field
        Slug = Seed.Posts().First().Slug
      };

      FluentActions.Invoking(() => SendAsync(command)).Should().ThrowAsync<ValidationException>();
    }

    [Test]
    public void GivenPostDoesNotExist_ThrowsNotFoundException()
    {
      var command = new CreateCommentCommand
      {
        Message = "Test comment",
        Slug = "does-not-exist"
      };

      async Task Handler() => await SendAsync(command);

      Assert.ThrowsAsync(typeof(NotFoundException), Handler);
    }

    [Test]
    public async Task GivenAValidRequest_CreatesComment()
    {
      var command = new CreateCommentCommand
      {
        Message = "Test comment",
        Slug = Seed.Posts().First().Slug,
      };

      var result = await SendAsync(command);
      var comment = await FindByIdAsync<Comment>(result.Id);

      Snapshot.Match(comment, options => options.IgnoreField("Id"));
    }
  }
}
