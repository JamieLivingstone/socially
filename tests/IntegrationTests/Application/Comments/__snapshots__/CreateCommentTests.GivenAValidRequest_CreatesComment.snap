{
  "Id": 5,
  "Message": "Test comment",
  "CreatedAt": "2022-03-18T11:33:49.013047Z",
  "AuthorId": 1,
  "Author": null,
  "PostId": 1,
  "Post": null
}
